src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/VsContainedLanguageComponentsFactoryBase.cs

```diff
@@ -49,6 +67,8 @@ internal abstract class VsContainedLanguageComponentsFactoryBase : IVsContainedL
             {
                 await ProjectServices.ThreadingService.JoinableTaskFactory.SwitchToMainThreadAsync();
 
+                await InitializeAsync().ConfigureAwait(true);
```

Comment:
In that case, should we be calling `SwitchToMainThreadAsync` in `InitializeAsync` as well, to ensure we preserve invariants? I believe that it's a very quick operation if already on the main thread, and if we ever have to call `InitializeAsync` from another method in this class it'll protect us from the exception.